src/vm/database/clarity_db.rs

```diff
@@ -680,6 +716,41 @@ impl<'a> ClarityDatabase<'a> {
             .expect("Failed to get block data.")
     }
 
+    /// Gets the `StacksBlockId` for the *current* chain tip Stacks block height, converts this
+    /// `StacksBlockId` to a `SortitionId` and then gets the `BurnchainHeaderHash` from the
+    /// `BurnStateDB` using this `SortitionId`.
+    ///
+    /// Returns Some() if `0 <= height < current_burnblock_height`, and None otherwise.
+    pub fn get_burnchain_block_header_hash_for_burnchain_height(
+        &mut self,
+        burnchain_block_height: u32,
+    ) -> Option<BurnchainHeaderHash> {
```

Comment:
@kantai @jcnelson can you review the logic in this method before I re-write the tests?

Reply:
This code does not work in the presence of Bitcoin forks.  The sortition DB and burnchain DB will process and store state for multiple different Bitcoin blocks at the same height, should a Bitcoin reorg happen.

If you want to query a burnchain block hash at a particular height, you need to know which burnchain fork you're on.  You can do this given the Stacks chain tip's ID block hash -- i.e. the ID block hash of the parent block that this Clarity code is building on top of.

The `id_bhh` variable you load below should be the _same ID hash_ that was passed to the inner `MarfedKV`'s `begin()` method as `current`.  This, in turn, is the ID hash passed as `current` to `ClarityInstance::begin_block()`.  You should *not* be trying to deduce `id_bhh` from a block height.